client/src/components/DataFiles/DataFilesModals/DataFilesCopyModal.jsx

```diff
@@ -13,61 +19,39 @@ import DataFilesProjectsList from '../DataFilesProjectsList/DataFilesProjectsLis
 const DataFilesCopyModal = React.memo(() => {
   const history = useHistory();
   const location = useLocation();
+  const { copy, setStatus } = useCopy();
+  const { getStatus, getProps, setProps, toggle: toggleModal } = useModal();
+
+  const { params } = useFileListing('FilesListing');
+  const {
+    params: modalParams,
+    data: files,
+    fetchListing,
+  } = useFileListing('modal');
 
   const dispatch = useDispatch();
-  const params = useSelector(
-    (state) => state.files.params.FilesListing,
-    shallowEqual
-  );
-  const modalParams = useSelector(
-    (state) => state.files.params.modal,
-    shallowEqual
-  );
 
   const reloadPage = () => {
+    fetchListing(modalParams);
     history.push(location.pathname);
-    dispatch({
-      type: 'FETCH_FILES_MODAL',
-      payload: { ...modalParams, section: 'modal' },
-    });
   };
-  const files = useSelector((state) => state.files.listing.modal, shallowEqual);
-  const isOpen = useSelector((state) => state.files.modals.copy);
-  const { showProjects, canMakePublic } = useSelector(
-    (state) => state.files.modalProps.copy
-  );
-  const status = useSelector(
-    (state) => state.files.operationStatus.copy,
-    shallowEqual
-  );
+
+  const isOpen = getStatus('copy');
+  const { showProjects, canMakePublic } = getProps('copy');
+
   const [disabled, setDisabled] = useState(false);
-  const systems = useSelector(
-    (state) => state.systems.storage.configuration,
-    shallowEqual
-  );
+  const { data: systems } = useSystems();
 
-  const selectedFiles = useSelector((state) =>
-    state.files.selected.FilesListing.map((i) => ({
-      ...state.files.listing.FilesListing[i],
-      id: uuidv4(),
-    }))
-  );
+  const { selectedFiles } = useSelectedFiles();
   const selected = useMemo(() => selectedFiles, [isOpen]);
 
-  const toggle = () =>
-    dispatch({
-      type: 'DATA_FILES_TOGGLE_MODAL',
-      payload: { operation: 'copy', props: {} },
-    });
+  const toggle = () => toggleModal({ operation: 'copy', props: {} });
 
   const onOpened = () => {
-    dispatch({
-      type: 'FETCH_FILES_MODAL',
-      payload: {
-        ...params,
-        path: '',
-        section: 'modal',
-      },
+    fetchListing({
+      api: 'tapis',
```

Comment:
Need to watch out for mismatch between opened system and selected system in the dropdown on modal open
<img width="1606" alt="Screen Shot 2022-04-06 at 2 50 38 PM" src="https://user-images.githubusercontent.com/20326896/162067585-5ef027b3-46ee-4625-b7bf-2896e8c6b7a1.png">

Reply:
Ugh, I swear I had this working before we started messing with the copy UX, I'll take another look.